src/geometry/isometry_alga.rs

```diff
@@ -85,12 +85,12 @@ where
 {
     #[inline]
     fn transform_point(&self, pt: &Point<N, D>) -> Point<N, D> {
-        self * pt
+        self.transform_point(pt)
```

Comment:
I adjusted the trait impls to delegate to the inherent impl, so that the algebra used for the inherent impl will stay synchronized between the two (e.g., if some of the `inverse_transform_point` implementations become optimized further.)